openmrs-keycloak-smart-auth/src/main/java/org/openmrs/contrib/keycloak/smart/auth/provider/SmartLaunchAuthenticator.java

```diff
@@ -110,6 +110,13 @@ private void selectPatient(AuthenticationFlowContext context) {
 		List<String> scopes = Arrays.stream(scope.split("\\s+")).collect(Collectors.toList());
 		String supportedParams = context.getAuthenticatorConfig().getConfig()
 				.get(SmartLaunchAuthenticatorFactory.CONFIG_EXTERNAL_SMART_LAUNCH_SUPPORTED_PARAMS);
+
+		if (scope.contains("launch/encounter")) {
+			patientSelectionUrl = patientSelectionUrl + "&launchType=encounter";
+		} else {
+			patientSelectionUrl = patientSelectionUrl + "&launchType=patient";
+		}
```

Comment:
@ibacher, thanks. Please correct me if I understood wrong. Do we need to pass the launchType (i.e., "encounter" or "patient") parameter in token securely?

Reply:
So two things:

1. Yes, if we are passing the launch type, it needs to be in the token
2. But, how are we going to address a request for multiple launch parameters? This can probably be handled either by multiple challenge steps in the authenticator or handled inside OMRS.

Reply:
I think handling it into the authenticator will be better because if we do this here, we only need to check the launchType in OMRS. 

So currently, we only support patient and encounter standalone launches. if the client request only for patient context (scope contain launch/patient but not launch/encounter) then we set the `launchType="patient"` but if client request for encounter context (scope contains both launch/patient and launch/encounter) then we set the `launchType="encounter"`.  By Checking the scope requested, we can easily predict the launchType. Thought?

Reply:
@ibacher Thought?